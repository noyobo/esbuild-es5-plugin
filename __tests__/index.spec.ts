import esbuild from 'esbuild';
import * as path from 'path';
import * as fse from 'fs-extra';

import es5Plugin from '../src/index';

const writeFileSync = (file: string, data: string) => {
  fse.ensureFileSync(file);
  fse.writeFileSync(file, data);
};

const emitHtml = async (files: string[]) => {
  files.forEach(file => {
    let output = file.replace(path.extname(file), '.html');
    output = path.relative(__dirname, output);
    output = path.join(__dirname, '../dist', output);

    const html = `<!doctype html>
<html lang='en'>
<head>
  <meta charset='UTF-8'>
  <meta name='viewport'
        content='width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0'>
  <meta http-equiv='X-UA-Compatible' content='ie=edge'>
  <title>file</title>
  <link rel="stylesheet" href="./index.css">
</head>
<body>
  <script src="./index.js"></script>
</body>
</html>
`;
    fse.ensureFileSync(output);
    fse.writeFileSync(output, html);
  });
};

const createBuild = async (files: string[]) => {
  files = files.map(file => {
    return path.join(__dirname, file);
  });

  await emitHtml(files);
  const result = await esbuild.build({
    entryPoints: files,
    plugins: [es5Plugin()],
    write: false,
    bundle: true,
    sourcemap: 'linked',
    outbase: __dirname,
    outdir: 'dist',
    target: ['es5'],
    minify: true,
  });

  for (const file of result.outputFiles) {
    if (file.path.endsWith('.js')) {
      expect(file.text).toMatchSnapshot();
      writeFileSync(file.path, file.text);
    }
  }
};

describe('index', () => {
  describe('esbuild-plugin-es5', () => {
    it('basic', async () => {
      await createBuild(['./fixtures/basic/index.ts']);
    });

    it('should support react', function () {
      return createBuild(['./fixtures/react/index.tsx']);
    });
  });
});
