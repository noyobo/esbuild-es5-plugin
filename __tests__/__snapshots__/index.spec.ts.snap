// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index esbuild-plugin-es5 async 1`] = `
""use strict";
import _sliced_to_array from "@swc/helpers/src/_sliced_to_array.mjs";
import { foo } from "./foo";
import { bar } from "./bar";
Promise.all([
  foo(),
  bar()
]).then(function(param) {
  var _param = _sliced_to_array(param, 2), _$foo = _param[0], _$bar = _param[1];
  console.warn(_$foo, _$bar);
});
//# sourceMappingURL=index.js.map
"
`;

exports[`index esbuild-plugin-es5 basic 1`] = `
""use strict";
var a = 1;
var b = "1 + 1";
console.log(a, b);
//# sourceMappingURL=index.js.map
"
`;

exports[`index esbuild-plugin-es5 error 1`] = `
[
  {
    "detail": undefined,
    "id": "",
    "location": null,
    "notes": [],
    "pluginName": "esbuild-plugin-es5",
    "text": "
  [38;2;255;30;30mÃ—[0m Expected ',', got ';'
   â•­â”€[[38;2;92;157;255;1;4m/Users/noyobo/home/github/esbuild-plugin-es5/__tests__/fixtures/error/index.ts[0m:1:1]
 [2m1[0m â”‚ function err(){
 [2m2[0m â”‚   console.log(error;
   Â· [38;2;246;87;248m                   â”€[0m
 [2m3[0m â”‚ }
   â•°â”€â”€â”€â”€


Caused by:
    Syntax Error",
  },
]
`;

exports[`index esbuild-plugin-es5 react 1`] = `
""use strict";
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _interopRequireDefault = require("@swc/helpers/lib/_interop_require_default.js").default;
var _react = /* @__PURE__ */ _interopRequireDefault(require("react"));
var _client = require("react-dom/client");
require("./index.css");
var App = function() {
  console.warn("Hello World");
  return /* @__PURE__ */ _react.default.createElement("div", null, "Hello World");
};
var div = document.createElement("div");
document.body.appendChild(div);
var root = (0, _client.createRoot)(div);
root.render(/* @__PURE__ */ _react.default.createElement(App, null));
//# sourceMappingURL=index.js.map
"
`;
